src/CounterInterceptor/Program.cs

```diff
@@ -0,0 +1,84 @@
+﻿namespace TemporalioSamples.CounterInterceptor;
+
+using Temporalio.Client;
+using Temporalio.Worker;
+
+internal class Program
+{
+    private static async Task Main(string[] args)
+    {
+        var client = await TemporalClient.ConnectAsync(
+            options: new("localhost:7233")
+            {
+                Interceptors = new[]
+                {
+                    new SimpleClientCallsInterceptor(),
+                },
+            });
+
+        using var tokenSource = new CancellationTokenSource();
+        Console.CancelKeyPress += (_, eventArgs) =>
+        {
+            tokenSource.Cancel();
+            eventArgs.Cancel = true;
+        };
+
+        var activities = new MyActivities();
+
+        var taskQueue = "CounterInterceptorTaskQueue";
+
+        var workerOptions = new TemporalWorkerOptions(taskQueue).
+                AddAllActivities(activities).
+                AddWorkflow<MyWorkflow>().
+                AddWorkflow<MyChildWorkflow>();
+
+        workerOptions.Interceptors = new[] { new SimpleCounterWorkerInterceptor() };

```

Comment:
You should consider combining these interceptors. Make the one interceptor implement both `IClientInterceptor` _and_ `IWorkerInterceptor`. The it's automatically used for worker interceptors and you don't have to set this setting. This is what we recommend here and is documented on the `Interceptors` field for client. See the https://github.com/temporalio/samples-dotnet/tree/main/src/ContextPropagation sample.